---
title: "liftover_pipeline"
author: "Briana Mittleman"
date: "8/1/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

I will use this analysis to create a pipeline I can use to liftover the peaks once I get them from the human and chimp three prime seq data.  

##Download files 


Tool to add to conda environment:  

* ucsc-liftover  

Chain file from UCSC:  

* /project2/gilad/briana/genome_anotation_data/comp_genomes/liftover/hg38ToPanTro5.over.chain.gz  

* /project2/gilad/briana/genome_anotation_data/comp_genomes/liftover/panTro5ToHg38.over.chain.gz  

##Prepare the bed files

I want the bed files with the peaks to be in the folowing format:  

chr# start end species_peakname

Resulting bed files will go in: /project2/gilad/briana/comparitive_threeprime/data/liftover

To go from the peak bed file created in the callPeaksbySpecies analysis I need to cut the file to the first four columns and add the species name to the peak.  

```{bash, eval=F}
awk '{print $1 "\t" $2 "\t" $3 "\t" "human_"$4}' /project2/gilad/briana/comparitive_threeprime/human/data/mergedPeaks_comb/filtered_APApeaks_merged_allchrom_named_human.bed > /project2/gilad/briana/comparitive_threeprime/data/liftover/filtered_humanPeaks.bed
```


```{bash, eval=F}
awk '{print $1 "\t" $2 "\t" $3 "\t" "chimp_"$4}' /project2/gilad/briana/comparitive_threeprime/chimp/data/mergedPeaks_comb/filtered_APApeaks_merged_allchrom_named_chimp.bed > /project2/gilad/briana/comparitive_threeprime/data/liftover/filtered_chimpPeaks.bed
```

##Run liftover  

Run liftOver with 'liftOver input.bed hg18ToHg19.over.chain.gz output.bed unlifted.bed' I want to run it both direction. I will then lift back.   


LiftForward.sh
```{bash, eval=F}
#!/bin/bash

#SBATCH --job-name=LiftForward
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=LiftForward.out
#SBATCH --error=LiftForward.err
#SBATCH --partition=broadwl
#SBATCH --mem=16G
#SBATCH --mail-type=END

module load Anaconda3
source activate comp_threeprime_env

#human to chimp
liftOver /project2/gilad/briana/comparitive_threeprime/data/liftover/filtered_humanPeaks.bed /project2/gilad/briana/genome_anotation_data/comp_genomes/liftover/hg38ToPanTro5.over.chain.gz /project2/gilad/briana/comparitive_threeprime/data/liftover/filtered_humanPeakslifted.bed /project2/gilad/briana/comparitive_threeprime/data/liftover/filtered_humanPeaksunlifted.bed
 

#chimp to human 
liftOver /project2/gilad/briana/comparitive_threeprime/data/liftover/filtered_chimpPeaks.bed /project2/gilad/briana/genome_anotation_data/comp_genomes/liftover/panTro5ToHg38.over.chain.gz /project2/gilad/briana/comparitive_threeprime/data/liftover/filtered_chimpPeaks.lifted.bed /project2/gilad/briana/comparitive_threeprime/data/liftover/filtered_chimpPeaks.unlifted.bed
```




LiftReverse.sh  

Now the lifted human peaks are on chimp cordinates and vise-versa. I will lift back over.  

```{bash, eval=F}
#!/bin/bash

#SBATCH --job-name=LiftReverse
#SBATCH --time=24:00:00
#SBATCH --output=LiftReverse.out
#SBATCH --error=LiftReverse.err
#SBATCH --partition=broadwl
#SBATCH --mem=16G
#SBATCH --mail-type=END

module load Anaconda3
source activate comp_threeprime_env


#human to chimp back to human
liftOver /project2/gilad/briana/comparitive_threeprime/data/liftover/filtered_humanPeakslifted.bed /project2/gilad/briana/genome_anotation_data/comp_genomes/liftover/panTro5ToHg38.over.chain.gz /project2/gilad/briana/comparitive_threeprime/data/liftover/filtered_humanPeakslifted_reverse.bed /project2/gilad/briana/comparitive_threeprime/data/liftover/filtered_humanPeaksunlifted.reverse.bed

#chimp to human back to chimp
liftOver /project2/gilad/briana/comparitive_threeprime/data/liftover/filtered_chimpPeaks.lifted.bed /project2/gilad/briana/genome_anotation_data/comp_genomes/liftover/hg38ToPanTro5.over.chain.gz /project2/gilad/briana/comparitive_threeprime/data/liftover/filtered_chimpPeaks.lifted.reverse.bed /project2/gilad/briana/comparitive_threeprime/data/liftover/filtered_chimpPeaks.unlifted.reverse.bed

```

LiftFinal.sh 

I now have lifted back and I want to go forward one more time to get the final list i should use.  


```{bash, eval=F}
#!/bin/bash

#SBATCH --job-name=LiftFinal
#SBATCH --time=24:00:00
#SBATCH --output=LiftFinal.out
#SBATCH --error=LiftFinal.err
#SBATCH --partition=broadwl
#SBATCH --mem=16G
#SBATCH --mail-type=END

module load Anaconda3
source activate comp_threeprime_env


#human to chimp back to human - final lift to chimp
liftOver /project2/gilad/briana/comparitive_threeprime/data/liftover/filtered_humanPeakslifted_reverse.bed /project2/gilad/briana/genome_anotation_data/comp_genomes/liftover/hg38ToPanTro5.over.chain.gz /project2/gilad/briana/comparitive_threeprime/data/liftover/filtered_humanPeakslifted_reverse.finalCcords.bed /project2/gilad/briana/comparitive_threeprime/data/liftover/filtered_humanPeaksunlifted.reverse.final.bed


#chimp to human back to chimp- final lift to human
liftOver /project2/gilad/briana/comparitive_threeprime/data/liftover/filtered_chimpPeaks.lifted.reverse.bed /project2/gilad/briana/genome_anotation_data/comp_genomes/liftover/panTro5ToHg38.over.chain.gz /project2/gilad/briana/comparitive_threeprime/data/liftover/filtered_chimpPeaks.lifted.reverse.finalHcords.bed /project2/gilad/briana/comparitive_threeprime/data/liftover/filtered_chimpPeaks.unlifted.reverse.final.bed

```


-change min percentage


The final lifted files have 350111 peaks that lifted from chimp to human and 442100 peaks that lifter human to chimp. I next will find the intersection of these files for the final list. In order to do this I need to create files that have the coordinates in human and in chimp. I can do this using the reverse file and final file.  


##Process results  

###human peaks  

```{r}
library(dplyr)
library(workflowr)
library(tidyverse)
library(reshape2)
```


```{r}
#human rev is in human coordinates 
human_rev= read.table("../data/liftover/filtered_humanPeakslifted_reverse.bed", col.names = c("human_chr", "human_start", "human_end", "name"), stringsAsFactors = F)

#final coords are in chimp coordinates  
human_lifted=read.table("../data/liftover/filtered_humanPeakslifted_reverse.finalCcords.bed", col.names = c("chimp_chr", "chimp_start", "chimp_end", "name"), stringsAsFactors = F)
```


I want to join these files by the name of the peaks keeping only the peaks that are in the final lifted.  

```{r}
human_final=human_lifted %>% left_join(human_rev, by="name")
```

###Chimp peaks  

```{r}
#chimp rev in chimp cords  

chimp_rev=read.table("../data/liftover/filtered_chimpPeaks.lifted.reverse.bed", col.names = c("chimp_chr", "chimp_start", "chimp_end", "name"), stringsAsFactors = F)

#final chimp lift is in human coords  

chimp_lifted=read.table("../data/liftover/filtered_chimpPeaks.lifted.reverse.finalHcords.bed", col.names=c( "human_chr", "human_start", "human_end", "name"),stringsAsFactors = F )
```


Join the files  

```{r}
chimp_final=chimp_lifted %>% left_join(chimp_rev, by="name")
```



###Intersection of reciprocal liftover  
```{r}
union_peaks=human_final %>% inner_join(chimp_final, by=c("human_chr", "human_start", "human_end", "chimp_chr", "chimp_start", "chimp_end" ))

peak_names=union_peaks %>% select(name.x, name.y) 
colnames(peak_names)= c("human", "chimp")
```
This leaves `r nrow(union_peaks)` 


I can then seperate these and write out the bedfile. With that I can look at metrics such as how many per gene or distance to last exon.  

```{r}
human_ortho= union_peaks %>% select(human_chr, human_start, human_end, name.x) 
chimp_ortho= union_peaks %>% select(chimp_chr, chimp_start, chimp_end, name.y)
```


Write these:  
```{r, eval=F}
write.table(human_ortho, file="../data/liftover/humanOrthoPeaks.bed", quote = F, row.names = F, col.names = F,sep="\t")
write.table(chimp_ortho, file="../data/liftover/chimpOrthoPeaks.bed", quote= F, row.names = F, col.names = F, sep="\t")
write.table(peak_names, file="../data/liftover/HumanChimpPeaknames.txt", quote=F, row.names = F, col.names = T, sep="\t")
```

##Plot results:  

I want to plot the number of peaks by proportion that are conserved similar to figure 1a in wang et al. 2018.  

```{r}
hTotal=456566
cTotal= 360860
hConprop= hTotal- 76207
cConprop=cTotal- 76207 
hUnConprop=hTotal - hConprop
cUnConprop=cTotal- cConprop

hum=c(hConprop,hUnConprop)
chi=c(cConprop, cUnConprop)
both=as.data.frame(rbind(hum, chi))
colnames(both)=c("Conserved", "Unconserved")
both=both %>% mutate( Species=c("Human", "Chimp"))
both_melt=melt(both, id.vars="Species")

pas_plot=ggplot(both_melt, aes(x=Species, fill=variable, y=value)) + geom_bar(stat="identity", position = "fill") + labs(y="Proportion PAS Peaks", title="Conserved vs Unconserved PAS") + scale_fill_discrete(name="Peak Class") + annotate("text", x=1, y=.85, label=paste("Total PAS:", hTotal, sep=" "))+ annotate("text", x=2, y=.85, label=paste("Total PAS:", cTotal, sep=" ")) + annotate("text", x=1, y=.15, label=paste("Conserved PAS:", "76207", sep=" ")) + annotate("text", x=2, y=.1, label=paste("Conserved PAS:", "76207", sep=" "))

pas_plot
```

